src/System.CommandLine/Binding/ModelDescriptor.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace System.CommandLine.Binding
+{
+    public class ModelDescriptor
```

Comment:
won't ModelDescriptor need to be a ValueDescriptor for properties that are complex types?

Reply:
Possibly. Complex types will be build up using their constituent constructor arguments and property setters, so there may not be a need for this.